src/query/sql/src/planner/binder/table.rs

```diff
@@ -427,21 +440,71 @@ impl Binder {
             .write()
             .add_base_column_scan_id(base_column_scan_id);
 
-        Ok((
-            SExpr::create_leaf(Arc::new(
-                Scan {
-                    table_index,
-                    columns: columns.into_iter().map(|col| col.index()).collect(),
-                    statistics: Arc::new(Statistics::default()),
-                    change_type,
-                    sample: sample.clone(),
-                    scan_id,
-                    ..Default::default()
-                }
-                .into(),
-            )),
+        let scan_s_expr = SExpr::create_leaf(Arc::new(
+            Scan {
+                table_index,
+                columns: columns.into_iter().map(|col| col.index()).collect(),
+                statistics: Arc::new(Statistics::default()),
+                change_type,
+                sample: sample.clone(),
+                scan_id,
+                ..Default::default()
+            }
+            .into(),
+        ));
+
+        // Check for row_access_policy and wrap with SecureFilter if present
+        let final_s_expr = if let Some(policy) =
+            &table.table().get_table_info().meta.row_access_policy
+        {
+            LicenseManagerSwitch::instance()
+                .check_enterprise_enabled(self.ctx.get_license_key(), Feature::RowAccessPolicy)?;
+            let meta_api = UserApiProvider::instance().get_meta_store_client();
+            let handler = get_row_access_policy_handler();
+            let res = databend_common_base::runtime::block_on(handler.get_row_access(
+                meta_api,
+                &self.ctx.get_tenant(),
+                policy.to_string(),
+            ))?;
+            let body = res.1.data.body;
+            let settings = self.ctx.get_settings();
+            let sql_dialect = settings.get_sql_dialect()?;
+            let tokens = tokenize_sql(&body)?;
```

Comment:
As far as I know, saving SQL as is in the meta is not safe because we cannot guarantee full compatibility.

Reply:
Maybe like view stmt, should use alter rewrite the the row access policy.